iox_arrow_flight/src/flightsql.rs

```diff
@@ -0,0 +1,161 @@
+//! Fork of the Arrow Rust FlightSQL client
+//! <https://github.com/apache/arrow-rs/tree/master/arrow-flight/src/sql/client.rs>
+//!
+//! Plan is to upstream much/all of this to arrow-rs
+//! see <https://github.com/apache/arrow-rs/issues/3301>
+
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use arrow_flight::sql::{CommandStatementQuery, ProstMessageExt};
+use arrow_flight::{FlightDescriptor, FlightInfo};
+use prost::Message;
+use tonic::metadata::MetadataMap;
+use tonic::transport::Channel;
+
+use crate::{
+    error::{FlightError, Result},
+    FlightClient, FlightRecordBatchStream,
+};
+
+/// A FlightSQLServiceClient handles details of interacting with a
+/// remote server using the FlightSQL protocol.
+#[derive(Debug)]
+pub struct FlightSqlClient {
+    inner: FlightClient,
+}
+
+/// An [Arrow Flight
+/// SQL](https://arrow.apache.org/docs/format/FlightSql.html) client
```

Comment:
```suggestion
/// An [Arrow Flight SQL](https://arrow.apache.org/docs/format/FlightSql.html) client
```
A minor nit, but the line break makes this hard to read imo